src/components/childComponents/FileMenu.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MenuIcon from '@mui/icons-material/Menu';
+import { downloadFile } from '../../utils/file';
+import {File} from "../../types/data";
+import {useState} from "react";
+import {fileDeleted} from "../../actions/file";
+import {useDispatch} from "react-redux";
+import {ActionModal} from "./ActionModal";
+import DownloadIcon from '@mui/icons-material/Download';
+import DeleteIcon from '@mui/icons-material/Delete';
+import {ListItemIcon, ListItemText} from "@mui/material";
+
+type FileMenuProps = {
+	file: File
+}
+
+export default function FileMenu(props: FileMenuProps) {
+	const {file} = props;
+	const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+	const open = Boolean(anchorEl);
+	const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+		setAnchorEl(event.currentTarget);
+	};
+	const handleClose = () => {
+		setAnchorEl(null);
+	};
+
+	// confirmation dialog
+	const dispatch = useDispatch();
+	const deleteFile = (fileId:string|undefined) => dispatch(fileDeleted(fileId));
+	const [confirmationOpen, setConfirmationOpen] = useState(false);
+	const deleteSelectedFile = () => {
+		if (file) {
+			deleteFile(file.id);
+		}
+		setConfirmationOpen(false);
+	}
+
+	return (
+		<div>
+			<ActionModal actionOpen={confirmationOpen} actionTitle="Are you sure?"
+						 actionText="Do you really want to delete? This process cannot be undone."
+						 actionBtnName="Delete" handleActionBtnClick={deleteSelectedFile}
+						 handleActionCancel={() => { setConfirmationOpen(false);}}/>
+			<Button
+				id="basic-button"
+				// variant="outlined"
+				size="small"
+				aria-controls={open ? 'basic-menu' : undefined}
+				aria-haspopup="true"
+				aria-expanded={open ? 'true' : undefined}
+				onClick={handleClick}
+			>
+				<MenuIcon/>
+			</Button>
+			<Menu
+				id="basic-menu"
+				anchorEl={anchorEl}
+				open={open}
+				onClose={handleClose}
+				MenuListProps={{
+					'aria-labelledby': 'basic-button',
+				}}
+			>
+				<MenuItem onClick={()=>{downloadFile(file.id, file.name);}}>
```

Comment:
Do you think we should add `handleClose();` inside the onClick action so that when you click download, the menu just collapses?

right now it is still present after you download.